ghproxy/ghcache/coalesce.go

```diff
@@ -50,15 +50,18 @@ type responseWaiter struct {
 // Notes: Deadlock shouldn't be possible because the map lock is always
 // acquired before responseWaiter lock if both locks are to be held and we
 // never hold multiple responseWaiter locks.
-func (r *requestCoalescer) RoundTrip(req *http.Request) (*http.Response, error) {
+func (r *requestCoalescer) RoundTrip(req *http.Request) (resp *http.Response, err error) {
 	// Only coalesce GET requests
 	if req.Method != http.MethodGet {
 		return r.delegate.RoundTrip(req)
 	}
 
 	var cacheMode = ModeError
 	defer func() {
-		cacheCounter.WithLabelValues(cacheMode).Inc()
+		cacheCounter.WithLabelValues(string(cacheMode)).Inc()
+		if resp != nil {
```

Comment:
This feels tricky and a bit fragile but if we test it well and don't touch this often it should be fine

Reply:
I got rid of the defer and named return value to make this less fragile.
Tests are also done now.